packages/auth/src/platform_browser/strategies/phone.ts

```diff
@@ -65,14 +78,92 @@ class ConfirmationResultImpl implements ConfirmationResult {
     );
     return this.onConfirmation(authCredential);
   }
+
+  async confirmWithWebOTP(
+    auth: Auth,
+    webOTPTimeout: number
+  ): Promise<UserCredential | undefined> {
+    if ('OTPCredential' in window) {
+      const abortController = new AbortController();
+      const timer = setTimeout(() => {
+        abortController.abort();
+
+        const myErr = _errorWithCustomMessage(
+          auth,
+          AuthErrorCode.WEB_OTP_NOT_RETRIEVED,
+          `Web OTP code is not fetched before timeout`
+        ) as WebOTPError;
+        myErr.confirmationResult = this;
+        throw myErr;
+      }, webOTPTimeout * 1000);
+
+      // @ts-ignore - ignore types for testing
```

Comment:
nit: remove this. IIRC, we tried this to avoid adding custom interfaces, but this line didnt help.